frontend/src/components/beta/accounts/AccountFungibleTokenHistory.tsx

```diff
@@ -0,0 +1,169 @@
+import { buildAccountUrl } from "../../../hooks/use-account-page-options";
+
+const TableWrapper = styled("div", {
+  display: "flex",
+  flexDirection: "column",
+  justifyContent: "center",
+  fontFamily: "Manrope",
+  paddingHorizontal: 24,
+  borderRadius: 8,
+  border: "1px solid #e8e8e8",
+  marginLeft: 20,
+  flex: 2,
+
+  "@media (max-width: 768px)": {
+    marginLeft: 0,
+    marginTop: 20,
+  },
+});
+
+const TableHeader = styled("thead", {
+  textTransform: "uppercase",
+  color: "#c4c4c4",
+  borderBottom: "1px solid #e8e8e8",
+
+  fontSize: 12,
+  fontWeight: 600,
+});
+
+const TableHeaderCell = styled("th", {
+  paddingVertical: 20,
+});
+
+const TableRow = styled("tr", {
+  fontSize: 14,
+  fontWeight: 500,
+  height: 50,
+
+  "& + &": {
+    borderTop: "1px solid #e8e8e8",
+  },
+});
+
+const TableElement = styled("td", {
+  verticalAlign: "top",
+  paddingVertical: 8,
+  paddingRight: 12,
+});
+
+const Link = styled(LinkWrapper, {
+  cursor: "pointer",
+
+  variants: {
+    disabled: {
+      true: {
+        cursor: "inherit",
+      },
+    },
+  },
+});
+
+type ItemProps = {
+  item: AccountFungibleTokenHistoryElement;
+  token: AccountFungibleToken;
+};
+
+const AccountFungibleTokenHistoryElementView: React.FC<ItemProps> = React.memo(
+  ({ item, token }) => {
+    const { t } = useTranslation();
+    return (
+      <TableRow>
+        <Link
+          href={
+            item.counterparty
+              ? buildAccountUrl({
+                  accountId: item.counterparty,
+                  tab: "fungible-tokens",
+                })
+              : ""
+          }
+          shallow
+          disabled={!item.counterparty}
+        >
+          <TableElement>
+            {item.counterparty
+              ? shortenString(item.counterparty)
+              : item.direction === "in"
+              ? "MINT"
+              : "BURN"}
+          </TableElement>
+        </Link>
+        <TableElement>
+          <TokenAmount
+            token={{ ...token, balance: item.amount }}
+            sign={item.direction === "out" ? -1 : 1}
+            noSymbol
+          />
+        </TableElement>
+        <Link
+          href={`/transactions/${item.transactionHash}#${item.receiptId}`}
+          shallow
+        >
+          <TableElement>{shortenString(item.receiptId)}</TableElement>
+        </Link>
+        <TableElement>
+          {moment(item.timestamp).format(t("common.date_time.date_format"))}
+        </TableElement>
+      </TableRow>
+    );
+  }
+);
+
+type Props = {
+  accountId: string;
+  token: AccountFungibleToken;
+};
+
+const AccountFungibleTokenHistory: React.FC<Props> = React.memo(
+  ({ accountId, token }) => {
+    const tokenHistoryQuery = trpc.useQuery([
+      "account-fungible-token-history",
+      { accountId, tokenAuthorAccountId: token.authorAccountId },
+    ]);
+    if (tokenHistoryQuery.status !== "success") {
+      if (tokenHistoryQuery.status === "loading") {
```

Comment:
Somehow I felt like it is more idiomatic to use UPPER_SNAKE_CASE for the string consts.

Reply:
That's `react-query` status constant.